### ISSUE TITLE
nodemon user.server  is not running successfully and is diplaying following error.

### ISSUE THREAD

**Issue by hafsazulf17** [socioboard/Socioboard-5.0#334]
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/www/socioboard/socioboard-api/User/core/unauthorized/unauthorized.routes.js' imported from /var/www/socioboard/socioboard-api/User/resources/routes/public.routes.js
    at new NodeError (internal/errors.js:322:7)
    at finalizeResolution (internal/modules/esm/resolve.js:318:11)
    at moduleResolve (internal/modules/esm/resolve.js:776:10)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:887:11)
    at Loader.resolve (internal/modules/esm/loader.js:89:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:76:40)
    at link (internal/modules/esm/module_job.js:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
[nodemon] app crashed - waiting for file changes before starting...

**Comment by BlackBlood22** [comment]
Yes this was one of the error i faced.the mistake is directory structure,in mycase it was of log folder which was ther as Log, but the code was looking for log  so it gave the error, i created a folder named log and that worked,not only user,but i had this problem in  3 diffrent modules,but no use after solving this issues when i tried to login when it comes to dashboard it crashes and i then just lost hopes and shutted down  the plan to install it

**Comment by hafsazulf17** [comment]
internal/process/esm_loader.js:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/www/socioboard/socioboard-api/Common/Models/unauthorized.model.js' imported from /var/www/socioboard/socioboard-api/User/core/unauthorized/unauthorized.service.js
    at new NodeError (internal/errors.js:322:7)
    at finalizeResolution (internal/modules/esm/resolve.js:318:11)
    at moduleResolve (internal/modules/esm/resolve.js:776:10)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:887:11)
    at Loader.resolve (internal/modules/esm/loader.js:89:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:76:40)
    at link (internal/modules/esm/module_job.js:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}


I got this error now

**Comment by arrianda** [comment]
#262 
You can follow step on that link and do the same with other uppercase error

**Comment by arrianda** [comment]
Maybe you have some config not written correctly? Do you find any log on Feeds/resources/Log?

**Comment by BlackBlood22** [comment]
@hafsazulf17  3 of the 5 services were having issue i solved it all manually and started the services but after that .I did the registration for a user.when i logged in the service just crashed adn stucked in dashboard no contents to view but tab is showing dashboard and the bar just keeps loading.

**Comment by arrianda** [comment]
Do you using socioboard web? Maybe you can try each function on swagger first, since dashboard on socioboard web maybe need all of the microservice to run smoothly

**Comment by hafsazulf17** [comment]
> Do you using socioboard web? Maybe you can try each function on swagger first, since dashboard on socioboard web maybe need all of the microservice to run smoothly



> @hafsazulf17 3 of the 5 services were having issue i solved it all manually and started the services but after that .I did the registration for a user.when i logged in the service just crashed adn stucked in dashboard no contents to view but tab is showing dashboard and the bar just keeps loading.


When I try to login after updating the database tables fields : 
```
use socioboarddb;
update user_activations set activation_status = 1;
```


`update user_activations set user_plan = 7;`


It goes to infinite loading and don't proceed further..

**Comment by hafsazulf17** [comment]
@sureshbabu1995 

I have folowing error in User Log file :
`error: TypeError: Cannot read property 'use_env_variable' of undefined       at file:///var/www/socioboard/socioboard-api/Common/Sequelize-cli/models/index.js`

If I run nodemon user.server.js It is stuck and not proceeding with success message Log file is created show the above error but If I run : systemctl status socioboard-user
It does not show me any error message and show successfully running at port 3000
If the URL when entered with :3000 is blank and does show anything but the service of node is running at port 3000

Also on Login to dashboard It shows me following message..

![image](https://user-images.githubusercontent.com/93780275/159483174-0b4e08e8-31ef-4442-a15f-71f90fe996ba.png)

**Comment by arrianda** [comment]
> @sureshbabu1995 @socioboard @arrianda @BlackBlood22
> 
> I am able to login to dashboard.. but errors when trying to add the social media accounts? ![linkedin](https://user-images.githubusercontent.com/93780275/159505329-34ba0e69-657e-4b99-9c5d-e46aed5019c7.JPG) ![facebook](https://user-images.githubusercontent.com/93780275/159505370-27a58521-f623-4fc4-bf98-e73b90ff38c5.JPG)

Do you already have valid and running social developer account? like Facebook Developer and it's API

**Comment by smokinjo** [comment]
> > @sureshbabu1995 @socioboard @arrianda @BlackBlood22
> > I am able to login to dashboard.. but errors when trying to add the social media accounts? ![linkedin](https://user-images.githubusercontent.com/93780275/159505329-34ba0e69-657e-4b99-9c5d-e46aed5019c7.JPG) ![facebook](https://user-images.githubusercontent.com/93780275/159505370-27a58521-f623-4fc4-bf98-e73b90ff38c5.JPG)
> 
> Do you already have valid and running social developer account? like Facebook Developer and it's API

Do we not just use our Facebook/LInkedIN/Instagram account?  When testing Socioboard paid version(free acccount), I just added my instagram account.   These are the accounts that we want to management.

**Comment by arrianda** [comment]
No, you need your own Developer Account for each social account you want to use if you want to use Socioboard on your own server/local. After that, you need to put correct setting for each Developer Account and config at each microservice /config/development.json. After that, you can freely connect any social media account to your deployed Socioboard

On paid version, it already use their own Socioboard's developer account which not shared because of legality and limit to each development account where you need to tell your exact website and callback.

**Comment by hafsazulf17** [comment]
@arrianda  @sureshbabu1995 
I see these two link for Instagram and Facebook developer accounts?
https://developers.facebook.com/async/registration/
https://www.instagram.com/accounts/login/?next=/developer/

Please mention more details about after creating account and generating API do we have to copy that API to each microservice config  file..

**Comment by sureshbabu1995** [comment]
HI @hafsazulf17 

I think you got it. Now I'm closing this ticket.

If you've any queries regarding this feel free to re-open it, if you have any other queries feel free to open a new ticket.

**Comment by hafsazulf17** [comment]
@arrianda  @sureshbabu1995 @BlackBlood22 

On login to dashboard It says :
Please upgrade your account 
How can we fix it from the backend currently the values are setup as:
`update user_activations set user_plan = 7;`

created at : 2022-03-19 15:05:55
account_expire_date : 2022-04-02 15:05:55
